### ISSUE TITLE
Genrate epub fail with KeyError: 'ids'

### ISSUE THREAD

**Issue by pprados** [sphinx-doc/sphinx#6370]
**Describe the bug**
When I produce an epub documentation, I receive an error.

**To Reproduce**
```
$ sphinx-build -M epub docs build_docs
```

**Expected behavior**

**Your project**
Link to your sphinx project, or attach zipped small project sample.

**Screenshots**

**Environment info**
- OS: Ubuntu 18.10
- Python version: 3.6.7
- Sphinx version: sphinx-build 2.0.1
- Sphinx extensions:  [
    'sphinx.ext.autodoc',
    'sphinx.ext.githubpages',
    'sphinx.ext.ifconfig',
    'sphinx.ext.mathjax',
    'sphinx.ext.napoleon',
    'sphinx.ext.todo',
    'sphinx.ext.viewcode',
    'm2r',
]

**Additional context**
Log file generated
[sphinx-err-xrvonbbm.log](https://github.com/sphinx-doc/sphinx/files/3178592/sphinx-err-xrvonbbm.log)

**Comment by tk0miya** [comment]
Could you share your project or create minimal reproducible example? I need it to fix the error.

**Comment by pprados** [comment]
Here, a sample of code.

[bdaproject.zip](https://github.com/sphinx-doc/sphinx/files/3182070/bdaproject.zip)

I use
sphinx-build -M epub docs build_docs

**Comment by tk0miya** [comment]
Thank you for example. I confirmed the error. And it was already fixed in 2.0 branch happily. Please wait for next release.
Thanks,

Note: Dockerfile to confirm the fixed in 2.0
```
FROM tk0miya/sphinx-html

RUN curl -LO https://github.com/sphinx-doc/sphinx/files/3182070/bdaproject.zip
RUN unzip bdaproject.zip
WORKDIR /docs/bdaproject
RUN apt install -y libffi-dev libssl-dev
RUN grep -v -e bdaproject requirements.txt > requirements2.txt
RUN pip3 install -r requirements2.txt
RUN pip3 install git+https://github.com/sphinx-doc/sphinx@2.0
RUN sphinx-build -M epub docs build_docs
```
